environments/gsm8k_server.py

```diff
@@ -232,14 +232,21 @@ async def collect_trajectories(
             "\\boxed{" + item["answer"].split("#")[-1].strip().replace(",", "") + "}"
         )
 
-        chat_completions = await self.server.chat_completion(
-            messages=[{"role": "system", "content": system_prompt}, user_message],
-            n=self.config.group_size,
-            max_tokens=self.config.max_token_length,
-        )
+        async with self.server.managed_server(tokenizer=self.tokenizer) as managed:
+
+            chat_completions = await managed.chat_completion(
+                messages=[{"role": "system", "content": system_prompt}, user_message],
+                n=self.config.group_size,
+                max_tokens=self.config.max_token_length,
+                temperature=1.0,
+            )
+
+            state = managed.get_state()
+            nodes = state["nodes"]
+
         to_score = list()
         to_backlog = list()
-        for i, chat_completion in enumerate(chat_completions.choices):
+        for i, chat_completion in zip(chat_completions.choices, nodes):
```

Comment:
this should still be enumerate, as the zip here gives you a tuple of (choice, node), and you're using i to index the node below.